watertap/unit_models/zero_order/CANDOP_zo.py

```diff
@@ -0,0 +1,83 @@
+"""
+
+from pyomo.environ import Var, units as pyunits
+from idaes.core import declare_process_block_class
+from watertap.core import build_sido_reactive, constant_intensity, ZeroOrderBaseData
+
+# Some more information about this module
+__author__ = "Travis Arnold"
+
+
+@declare_process_block_class("CANDOPZO")
+class CANDOPData(ZeroOrderBaseData):
+    """
+    Zero-Order model for a CANDO+P reactor unit.
+    """
+
+    CONFIG = ZeroOrderBaseData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        self._tech_type = "CANDO_P"
+
+        build_sido_reactive(self)
+
+        # Create electricity variable
+        self.electricity = Var(self.flowsheet().time,
+                               units=pyunits.kW,
+                               bounds=(0, None),
+                               doc="Electricity consumption of unit")
+        self._perf_var_dict["Electricity Demand"] = self.electricity
+
+        # Create electricity intensity variable and constraint. For this
+        # model, electricity demand is calculated based on the amount of
+        # nitrogen reacted.
+        # TODO The information I have says that this electricity consumption
+        #      accounts pumping, oxygenation, and stirring. At some point,
+        #      perhaps we should come back and adjust the model to account for
+        #      pumping costs separately.
+        self.electricity_intensity_N = Var(units=pyunits.kWh/pyunits.kg,
+                                           bounds=(0, None),
+                                           doc="Electricity demand per kg N reacted")
+        self._fixed_perf_vars.append(self.electricity_intensity_N)
+        self._perf_var_dict["Electricity Intensity"] = self.electricity_intensity_N
+        @self.Constraint(self.flowsheet().time,
+                         doc='Constraint for electricity consumption based on '
+                             'nitrogen consumed.')
+        def electricity_consumption(b, t):
+            return b.electricity[t] == (pyunits.convert(b.extent_of_reaction[t, "n_reaction"] * b.electricity_intensity_N, to_units=pyunits.kW))
+
+        # Create oxygen demand variables and constraint. The amount of oxygen
+        # consumed is assumed to be a linear function of the amount of
+        # nitrogen reacted.
+        self.O2_demand = Var(self.flowsheet().time,
+                             units=pyunits.kg/pyunits.s,
+                             bounds=(0, None),
+                             doc="Oxygen demand")
+        self._perf_var_dict["Oxygen Demand"] = self.O2_demand
+        self.oxygen_nitrogen_ratio = Var(units=pyunits.dimensionless,
+                                         bounds=(0, None),
+                                         doc="Oxygen consumed / nitrogen reacted ratio")
+        self._fixed_perf_vars.append(self.oxygen_nitrogen_ratio)
+        self._perf_var_dict["Oxygen consumed / nitrogen reacted ratio (mass basis)"] = self.oxygen_nitrogen_ratio
```

Comment:
Probably don't need to report this ratio if this is just an intermediate variable.

Reply:
It also doesn't hurt to show it. I guess it comes down to preference. For me, I'd prefer to include it.

Reply:
I included `oxygen_nitrogen_ratio` in the report because I thought it might be useful for the end user to see. I included `electricity_intensity_N` in the report as well for the same reason, at @andrewlee94 's suggestion. My inclination is to keep it as is, I think it's a helpful piece of information to display to the user.